iml-agent/src/action_plugins/ldev.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+use iml_wire_types::LdevEntry;
+use tokio::{fs::File, io::AsyncWriteExt};
+
+async fn write_to_file(content: String) -> Result<(), ImlAgentError> {
+    let ldev_path = env::get_var("LDEV_CONF_PATH");
+
+    let mut file = File::create(ldev_path).await?;
+    file.write_all(content.as_bytes()).await?;
+
+    Ok(())
+}
+
+async fn create_internal<F>(
+    entries: Vec<LdevEntry>,
+    write_to_file: impl Fn(String) -> F,
+) -> Result<(), ImlAgentError>
```

Comment:
This fn is more complex than needed.

If you called it something like `convert`, you could pass in entries and return a `String`. Then you would not need to make this a higher-order function, and could still test it in isolation.